test/postControllerTest.js

```diff
@@ -0,0 +1,148 @@
+/* global it, describe */
+const express = require('express');
+const request = require('supertest');
+const postController = require('../controllers/postController');
+const sinon = require('sinon');
+const postService = require('../services/postService');
+const assert = require('assert');
+
+const app = express();
+
+app.set('view engine', 'ejs');
+app.use(express.static('../public'));
+
+app.use('', postController);
+
+describe('postController', () => {
+  beforeEach(() => {
+    sinon.stub(postService, 'getAllPosts').resolves('expected');
+    sinon.stub(postService, 'getOnePost').resolves('expected');
+    sinon.stub(postService, 'createNewPost').resolves('expected');
+  });
+
+  afterEach(() => {
+    postService.getAllPosts.restore();
+    postService.getOnePost.restore();
+    postService.createNewPost.restore();
+  });
+
+  describe('GET /', () => {
+    it('should call method "postService.getAllPosts" with no arguments', () => (
+      request(app)
+        .get('/')
+        .then(() => {
+          sinon.assert.calledWith(postService.getAllPosts);
+        })
+    ));
+  });
+  describe('GET /', () => {
+    it('should return correct "posts" in promise', () => (
+      request(app)
+        .get('/')
+        .then(() => {
+          postService.getAllPosts().then((posts) => {
+            assert.equal(posts, 'expected');
```

Comment:
For clarity (since the other comment chain is marked as out-dated): This is still testing `sinon`, and doesn't actually test anything.

You can confirm this by deleting the following from postController.js and this test will still pass.
```
router.get(['/', '/posts'], (req, res) => {
  postService.getAllPosts().then((posts) => {
    res.render('posts', { posts });
  }).catch((err) => {
-    res.status(503).send(err.message);
  });
});
```